static/src/javascripts/projects/common/modules/commercial/slice-adverts.js

```diff
@@ -1,120 +1,118 @@
-                addFabricAd  = (config.page.isFront && config.switches.fabricAdverts && detect.isBreakpoint({max : 'phablet'}));
+    return {
+        init: init
+    };
 
-            // insert fabric advert at first container in lieu of a top slot
-            if (addFabricAd) {
-                fabricAdSlot = bonzo(createAdSlot('fabric', 'container-inline'));
-                fabricAdSlot.addClass('ad-slot--mobile');
-                fabricAdSlot.insertAfter(containers[0]);
-            }
+    function init() {
+        if (!commercialFeatures.sliceAdverts) {
+            return false;
+        }
 
-            // skip the initial containers if we've already added an advert to the first
-            index+= addFabricAd ? (1 + containerGap) : 0;
+        var fabricAdSlot, adSlots;
+        var prefs          = userPrefs.get('container-states') || {};
+        var isMobile       = detect.getBreakpoint() === 'mobile';
+        var isNetworkFront = ['uk', 'us', 'au'].indexOf(config.page.pageId) !== -1;
+        var hasFabricAd    = (config.page.isFront && config.switches.fabricAdverts && detect.isBreakpoint({max : 'phablet'}));
 
+        adSlots = qwery(containerSelector)
+            // get all ad slices
+            .map(function (container) {
+                return {
+                    container: container,
+                    adSlice: container.querySelector(sliceSelector)
+                };
+            })
             // pull out ad slices which have at least x containers between them
-            while (index < containers.length) {
-                container    = containers[index];
-                containerId  = bonzo(container).data('id');
-                $adSlice     = $(opts.sliceSelector, container);
-                // don't display ad in the first container on the network fronts
-                isFrontFirst = contains(['uk', 'us', 'au'], config.page.pageId) && index === 0;
-
+            .filter(function (item, index) {
                 if (config.page.showMpuInAllContainers) {
-                    adSlices.push($adSlice.first());
-                    index++;
-                } else {
-                    if ($adSlice.length && !isFrontFirst && (!prefs || prefs[containerId] !== 'closed')) {
-                        adSlices.push($adSlice.first());
-                        index += (containerGap + 1);
-                    } else {
-                        index++;
-                    }
+                    return true;
                 }
-            }
 
-            return Promise.all(chain(adSlices).slice(0, maxAdsToShow).and(map, function ($adSlice, index) {
-                    // When we are inside the AB test we are adding inline1 manually so index needs to start from 2.
-                    var inlineIndexOffset = (config.tests.cmTopBannerPosition) ? 2 : 1;
+                // Slots are inserted every containerGap + 1 slice
+                if (!item.adSlice || index % (containerGap + 1) !== 0) {
```

Comment:
This isn't quite the same as the original behaviour. Here, we're taking all odd-numbered containers that have advert candidates. The original would find slots with candidates, then skip scanning until `containerGap` items afterwards.

What this means (unless I've missed a subtlety of the logic) is that if only even-numbered containers had slice-candidates, the original code would display those adverts, but our new code would not display any of them.

e.g.
````
Index     Container     Old     New     What stops
          has slice?                    ad?

0         X             X       X       !item.adSlice
1         √             √       X       (1 % 2) !== 0
2         X             X       X       !item.adSlice
3         √             √       X       (3 % 2) !== 0
````